planner/aamwriter/src/main/java/eu/seaclouds/platform/planner/aamwriter/Translator.java

```diff
@@ -297,9 +297,8 @@ private String buildArtifactName(DNode dnode, boolean needsContainer) {
         return artifactName;
     }
     
-    private boolean isSource(DGraph dgraph, DNode dnode) {
-        List<DLink> links = dgraph.getLinksTo(dnode);
-        return links.size() == 0;
+    private boolean isFrontend(DGraph dgraph, DNode dnode) {
+        return dnode.equals(dgraph.getFrontendNode());

```

Comment:
MINOR: Perhaps this could be a method of "DNode", something like "dnode.isFrontend(dgraph)".